packages/hardhat-ethers/src/signers.ts

```diff
@@ -1,13 +1,30 @@
 import { ethers } from "ethers";
 
+// This is taken from @ethersproject/abstract-signer package.
+// EIP-712 Typed Data
+// See: https://eips.ethereum.org/EIPS/eip-712
+
+interface TypedDataDomain {
+  name?: string;
+  version?: string;
+  chainId?: ethers.BigNumberish;
+  verifyingContract?: string;
+  salt?: ethers.BytesLike;
+}
+
+interface TypedDataField {
+  name: string;
+  type: string;
+}
```

Comment:
I would replace this with:

```
type SignTypedDataParams = Parameters<ethers.providers.JsonRpcSigner['_signTypedData']>
type TypedDataDomain = SignTypedDataParams[0]
type TypedDataFields = SignTypedDataParams[1]
```

Reply:
In fact, we can use only the first line and then do:

```
  public _signTypedData(...args: SignTypedDataParams): Promise<string> {
    return this._signer._signTypedData(...args);
  }
```

below.

Reply:
I actually attempted to do this at first but it wouldn't recognize the `_signTypedData` member for some reason. It might have been my IDE Typescript integration acting up though, I don't think I tried actually building it.

Reply:
I went ahead and implemented it, without creating a new type, but rather making it explicit that it receives the same params.